src/bacnet/basic/object/bi.c

```diff
@@ -615,3 +861,417 @@ bool Binary_Input_Write_Property(BACNET_WRITE_PROPERTY_DATA *wp_data)
+    } else
+        return -1; /* end of list  */
+
+    if ((IsActiveEvent) || (IsNotAckedTransitions)) {
+        /* Object Identifier */
+        getevent_data->objectIdentifier.type = OBJECT_BINARY_INPUT;
+        getevent_data->objectIdentifier.instance =
+            Binary_Input_Index_To_Instance(index);
+        /* Event State */
+        getevent_data->eventState = BI_Descr[index].Event_State;
+        /* Acknowledged Transitions */
+        bitstring_init(&getevent_data->acknowledgedTransitions);
+        bitstring_set_bit(&getevent_data->acknowledgedTransitions,
+            TRANSITION_TO_OFFNORMAL,
+            BI_Descr[index]
+                .Acked_Transitions[TRANSITION_TO_OFFNORMAL]
+                .bIsAcked);
+        bitstring_set_bit(&getevent_data->acknowledgedTransitions,
+            TRANSITION_TO_FAULT,
+            BI_Descr[index].Acked_Transitions[TRANSITION_TO_FAULT].bIsAcked);
+        bitstring_set_bit(&getevent_data->acknowledgedTransitions,
+            TRANSITION_TO_NORMAL,
+            BI_Descr[index].Acked_Transitions[TRANSITION_TO_NORMAL].bIsAcked);
+        /* Event Time Stamps */
+        for (i = 0; i < 3; i++) {
+            getevent_data->eventTimeStamps[i].tag = TIME_STAMP_DATETIME;
+            getevent_data->eventTimeStamps[i].value.dateTime =
+                BI_Descr[index].Event_Time_Stamps[i];
+        }
+        /* Notify Type */
+        getevent_data->notifyType = BI_Descr[index].Notify_Type;
+        /* Event Enable */
+        bitstring_init(&getevent_data->eventEnable);
+        bitstring_set_bit(&getevent_data->eventEnable, TRANSITION_TO_OFFNORMAL,
+            (BI_Descr[index].Event_Enable & EVENT_ENABLE_TO_OFFNORMAL) ? true
+                                                                       : false);
+        bitstring_set_bit(&getevent_data->eventEnable, TRANSITION_TO_FAULT,
+            (BI_Descr[index].Event_Enable & EVENT_ENABLE_TO_FAULT) ? true
+                                                                   : false);
+        bitstring_set_bit(&getevent_data->eventEnable, TRANSITION_TO_NORMAL,
+            (BI_Descr[index].Event_Enable & EVENT_ENABLE_TO_NORMAL) ? true
+                                                                    : false);
+        /* Event Priorities */
+        Notification_Class_Get_Priorities(
+            BI_Descr[index].Notification_Class, getevent_data->eventPriorities);
+
+        return 1; /* active event */
+    } else
+        return 0; /* no active event at this index */
+}
+
+
+bool Binary_Input_Alarm_Value_Set(
+    uint32_t object_instance, BACNET_BINARY_PV value)
+{
+    unsigned index = 0;
+    bool status = false;
+
+    index = Binary_Input_Instance_To_Index(object_instance);
+    if (index < BI_Max_Index) {
+        if (Polarity[index] != POLARITY_NORMAL) {
+            if (value == BINARY_INACTIVE) {
+                value = BINARY_ACTIVE;
+            } else {
+                value = BINARY_INACTIVE;
+            }
```

Comment:
alternatively, feel free to disregard
value = (value == BINARY_INACTIVE) ? BINARY_ACTIVE : BINARY_INACTIVE;